scripts/dns-zone-and-server-certificates/create-or-renew-one-certificate.sh

```diff
@@ -0,0 +1,131 @@
+  echo ""
+  echo "This script require following executable configured in the PATH variable:"
+  echo " - aws cli 2.0 "
+  echo " - jq"
+  echo " - sha256"
+
+  if ( [ "$BASH_SOURCE" = "" ] ) then
+    return 1
+  else
+    exit 1
+  fi
+fi
+
+domainName=$1
+zoneName=$2
+profile=$3
+certificateRegion=$4
+zoneRegion=$5
+
+fullDomain="${domainName}.${zoneName}"
+
+
+certificateArn=$( aws acm --profile $profile --region $certificateRegion list-certificates \
+    | jq ".CertificateSummaryList[] | select(.DomainName==\"${fullDomain}\") | .CertificateArn" )
+
+if ( [ "" = "$certificateArn" ] ) then
+  echo "Do not exist a certificate for $fullDomain, require one"
+  aws acm --profile $profile --region $certificateRegion request-certificate \
+        --domain-name $fullDomain \
+        --validation-method DNS
+
+  echo "Give 5 second to request creation"
+  sleep 5
+
+  certificateArn=$( aws acm --profile $profile --region $certificateRegion list-certificates \
+    | jq ".CertificateSummaryList[] | select(.DomainName==\"${fullDomain}\") | .CertificateArn" )
+  if ( [ "" = "$certificateArn" ] ) then
+    echo "!!!! CANNOT CREATE A CERTIFICATE REQUEST"
+
+    if ( [ "$BASH_SOURCE" = "" ] ) then
+      return 1
+    else
+      exit 1
+    fi
+  fi
+fi
+
+# clean " character derived from json representation
+certificateArn=$( echo $certificateArn | tr -d '"' )
+
+echo ""
+echo "Certificate for ${fullDomain} has ARN ${certificateArn}"
+
+certificateDescription=$( aws acm --profile $profile --region $certificateRegion describe-certificate \
+                                  --certificate-arn $certificateArn )
+
+echo "Certificate Description"
+echo "$certificateDescription"
+
+certificateStatus=$( echo "$certificateDescription" | jq '.Certificate.Status' | tr -d '"' )
+
+echo "Certificate status = $certificateStatus"
+
+
+if ( [ "PENDING_VALIDATION" = "$certificateStatus" ] ) then
+  echo "Certificate needs validation"
+  validationDnsName=$( echo "$certificateDescription" | jq '.Certificate.DomainValidationOptions[0].ResourceRecord.Name' | tr -d '"' )
+  validationDnsType=$( echo "$certificateDescription" | jq '.Certificate.DomainValidationOptions[0].ResourceRecord.Type' | tr -d '"' )
+  validationDnsValue=$( echo "$certificateDescription" | jq '.Certificate.DomainValidationOptions[0].ResourceRecord.Value' | tr -d '"' )
+
+  echo "validation needs DNS entry"
+  echo " - Name: $validationDnsName"
+  echo " - Type: $validationDnsType"
+  echo " - Value: $validationDnsValue"
+
+  echo "{ \"Changes\": [{ \n"\
+       "    \"Action\": \"CREATE\", \n"\
+       "    \"ResourceRecordSet\": { \n"\
+       "        \"Name\": \"${validationDnsName}\",\n"\
+       "        \"Type\": \"${validationDnsType}\",\n"\
+       "        \"TTL\": 300,"\
+       "        \"ResourceRecords\": [{ \"Value\": \"${validationDnsValue}\"}] \n"\
+       "    }\n"\
+       "}]}" \
+        > /tmp/route53RecordSetChanges.json
```

Comment:
Importanza < 0 ma per i file temporanei uso sempre tmpfile in modo che non ci possono essere "sovrapposizioni" nel caso una precedente esecuzione abbia lasciato un file "sporco"

Reply:
In questo caso ho proprio l'intenzione di sovrascrivere file prodotti da precedenti esecuzioni.